src/common/post.ts

```diff
@@ -252,3 +257,80 @@ export const validatePost = (
 export const submitArticleThreshold = parseInt(
   process.env.SUBMIT_ARTICLE_THRESHOLD,
 );
+
+export const insertCodeSnippets = async ({
+  entityManager,
+  post,
+  codeSnippetsJson,
+}: {
+  entityManager: EntityManager;
+  post: Pick<Post, 'id'>;
+  codeSnippetsJson: PostCodeSnippetJsonFile;
+}) => {
+  const codeSnippets = codeSnippetsJson.snippets.reduce(
+    (acc, codeSnippetContent, index) => {
+      const checksum = createHash('sha1');
+      checksum.update(codeSnippetContent);
+      const contentHash = checksum.digest('hex');
+
+      if (!acc.has(contentHash)) {
+        acc.set(
+          contentHash,
+          entityManager.getRepository(PostCodeSnippet).create({
+            postId: post.id,
+            contentHash,
+            order: index,
+            content: codeSnippetContent,
+          }),
+        );
+      }
+
+      return acc;
+    },
+    new Map<string, PostCodeSnippet>(),
+  );
+
+  const uniqueCodeSnippets = Array.from(codeSnippets.values());
+
+  await entityManager.getRepository(PostCodeSnippet).delete({
+    postId: post.id,
+    contentHash: Not(
+      In(uniqueCodeSnippets.map((snippet) => snippet.contentHash)),
+    ),
+  });
```

Comment:
no because there could be code snippets that are removed from the article, see https://github.com/dailydotdev/daily-api/pull/2188#discussion_r1736384035

Reply:
cool cool :)